src/main/java/org/cactoos/text/Sub.java

```diff
@@ -79,7 +79,7 @@ public Sub(final Text text, final Scalar<Integer> strt) {
      * @param text The Text
      * @param strt Start position in the text
      */
-    public Sub(final Text text, final Func<String, Integer> strt) {
+    public Sub(final Text text, final Func<? super String, Integer> strt) {
```

Comment:
@DmitryBarskov Is `? super String` correct here?

Reply:
@andreoss I think so.

Having `? super String` makes it possible to pass both `Func<String, Integer>` (as in current implementation, leaving backward compatibility) and `Func<CharSequence, Integer>` types.

Reply:
@DmitryBarskov here too please add a test to back this assertion